providers/gitlab.go

```diff
@@ -64,6 +105,15 @@ func (p *GitLabProvider) Redeem(ctx context.Context, redirectURL, code string) (
 	return
 }
 
+// SetProjectScope ensure read_api is added to scope when filtering on projects
+func (p *GitLabProvider) SetProjectScope() {
+	if len(p.Projects) > 0 {
+		if !strings.Contains(p.Scope, "read_api") {
```

Comment:
NIT: This logic isn't aware of space delimiters. So if for some reason had a claim of `read_api_WEIRD` this would think `read_api` is present and not add the scope you intend.

Not sure if this is a bug worth fixing through.